src/coreclr/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs

```diff
@@ -221,7 +253,7 @@ private static object PtrToStructureHelper(IntPtr ptr, Type structureType)
         [MethodImpl(MethodImplOptions.InternalCall)]
         internal static extern bool IsPinnable(object? obj);
 
-#if FEATURE_COMINTEROP
+#if TARGET_WINDOWS
```

Comment:
You may want to also:
- Ifdef `QCFuncElement("GetHINSTANCE", COMModule::GetHINSTANCE)` in ecalllist.h for Windows
- Move the fallback implementation from src\libraries\System.Private.CoreLib\src\System\Runtime\InteropServices\Marshal.NoCom.cs to src\libraries\System.Private.CoreLib\src\System\Runtime\InteropServices\Marshal.cs, and put if under `#if !CORECLR && !TARGET_WINDOWS` to make it clear that the CoreCLR has special implementation for Windows, but the method has nothing to do with COM otherwise

Reply:
Reverted here and opened https://github.com/dotnet/runtime/pull/51523.